packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -1251,32 +1251,39 @@ export function resolveClassComponentProps(
 ): Object {
   let newProps = baseProps;
```

Comment:
You can just be explicit here with the types, `Object` is just an alias for `any`.

```suggestion
  let newProps: any = baseProps;
```

Reply:
Then the `typeof newProps` below can be removed.